python/ray/ml/predictors/integrations/torch/torch_predictor.py

```diff
@@ -82,7 +82,11 @@ def _convert_to_tensor(
     def _predict(self, tensor: torch.Tensor) -> pd.DataFrame:
         """Handle actual prediction."""
         prediction = self.model(tensor).cpu().detach().numpy()
-        return pd.DataFrame(prediction, columns=["predictions"])
+        # If model has outputs a Numpy array (for example outputting logits),
+        # these cannot be used as values in a Pandas Dataframe.
+        # We have to convert the outermost dimension to a python list (but the values
+        # in the list can still be Numpy arrays).
+        return pd.DataFrame({"predictions": list(prediction)}, columns=["predictions"])
```

Comment:
Can we use `TensorArray` here now that #24752 is merged?

```suggestion
        return pd.DataFrame({"predictions": TensorArray(prediction)}, columns=["predictions"])
```

Reply:
I didn't want to have a ray.data dependency in Predictors. Once we do the data type conversion cleanup and probably move `TensorArray` to a common utils file, I definitely agree though, we should use TensorArray here!